### PR TITLE
Integrate WooCommerce default filter by price widget with ES range query 

### PR DESCRIPTION
**Pull request by oscarssanchez**
### Description of the Change

This PR modifies EP default query for a range query (https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html) when the default WooCommerce filter by price widget has been set and submitted.

### Alternate Designs

I explored the possibility of building an EP widget for this purpose but came into conclusion accommodating for the range query was the only necessary step. This also enables users to create alternative designs for filtering e.g a text input

### Benefits

I've seen filtering has been a pain point for ElasticPress users as well as ElasticPress.io users. This would be a step forward to provide support for advanced filtering with the use of facets. 

### Possible Drawbacks

Might need testing in more scenarios.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by traed** [comment]
@oscarssanchez I did some testing against WooCommerce default products. There seem to be some issues though. If I filter on prices between 0 and 15, products that cost 2 disappears. If I filter between 0 and 2, products that cost between 0 and 19 appears. My guess is that it compares strings when it should be comparing numeric values.

**Comment by oscarssanchez** [comment]
Hi @traed ,

Thanks so much for spotting that bug. I can confirm in the range between 11-19 elasticsearch is not returning any results for $2 products. The mappings of meta._price are set to double not to string, so I don't think this is iikely the issue. 

I could not reproduce the bug where if you filter between 0-2 products that cost more appear. On that example, i get it to show only the $2 product one. 

I reached out in elasticsearch forums in regard to see if there's any insight.

**Comment by oscarssanchez** [comment]
@traed turns out i was looking at meta._value.price instead of meta._long.price, after this it looks like this is working correctly. Could you test one more time ? cc @allan23

**Comment by traed** [comment]
@oscarssanchez Seems to be working now :)

**Comment by oscarssanchez** [comment]
adding @tlovett1 for review

File: includes/classes/Feature/WooCommerce/WooCommerce.php [hunk 1]
```diff
@@ -824,4 +825,53 @@ public function requirements_status() {
...
+	 * @return mixed
+	 *
+	 */
+	public function price_filter( $args, $query_args ) {
+		$set_low_price = ! empty( $_GET['min_price'] );
+		$set_high_price = ! empty( $_GET['max_price'] );
+		$query = $args['query'];
+
+		//Account for match all context
+		if ( ( $set_high_price || $set_low_price ) && ! empty( $args['query']['match_all'] ) ) {
```
Comment (tlovett1) [inline includes/classes/Feature/WooCommerce/WooCommerce.php]: Will throw a notice/warning if `query` not set.

File: includes/classes/Feature/WooCommerce/WooCommerce.php [hunk 2]
```diff
@@ -824,4 +825,53 @@ public function requirements_status() {
...
+			unset ( $args['query']['match_all'] );
+			$args['query']['range']['meta._price.long']['gte'] = $_GET['min_price'];
```
Comment (tlovett1) [inline includes/classes/Feature/WooCommerce/WooCommerce.php]: Will throw a warning/notice if many of the parent array properties are not set.

File: includes/classes/Feature/WooCommerce/WooCommerce.php (hunk 1 above)
Comment (oscarssanchez) [reply includes/classes/Feature/WooCommerce/WooCommerce.php]: Thanks @tlovett1 I will address this on next commit.

File: includes/classes/Feature/WooCommerce/WooCommerce.php (hunk 2 above)
Comment (oscarssanchez) [reply includes/classes/Feature/WooCommerce/WooCommerce.php]: Hi @tlovett1, sorry I did not understand this comment, for what i know `$args['query']['range']['meta._price.long']['gte'] = $_GET['min_price'];` will create all of the parent array properties and `unset ( $args['query']['match_all'] );` is getting checked on L.843 

Could you please outline the steps to reproduce the scenario where this is failing so I get a better idea of what is wrong here?

**Comment by oscarssanchez** [comment]
Thanks for your comments @tlovett1 , I had one question in regard of one of them

**Comment by oscarssanchez** [comment]
Hi @tlovett1 , i updated the PR to check if $args['query'] is empty. 

For the checks requested for https://github.com/10up/ElasticPress/pull/1342/commits/9f913d5bd6792d83a5a265f1fd1681a47e1eb9f5#diff-398dcfe6a8fd4b7046b31cadddc5e6bbR852 and similar other ones, my intention here was to create all the parent arrays. So for example, for `$args['query']['range']['meta._price.long']['lte']`, the parent `$args['query']['range']` will never be set prior to the variable assignment. This is not throwing an error for me and creates the parent and value just fine:

This is $args['query'] before:

es2


And after parent array creation/unsetting match all array:

es1


The same can be achieved by creating the `['range']['meta._price.long']['lte']` array and then merging it with `$args['query']` but I thought this was easier. I can adjust if this approach makes it easier to understand for others.

**Comment by oscarssanchez** [comment]
Fixed the phpcs errors now so CI build passes.

**Comment by tlovett1** [comment]
See #1491

**Pull request closed without merging**